packages/story-editor/src/output/utils/getStoryMarkup.js

```diff
@@ -33,11 +33,22 @@ import OutputStory from '../story';
  * @return {string} Story markup.
  */
 export default function getStoryMarkup(story, pages, metadata) {
-  // Note that react-dom/server will warn about useLayoutEffect usage here.
-  // Not because of any wrongdoing in our code, but mostly because
-  // of its own profiler.
-  // See https://github.com/facebook/react/issues/14927
-  return renderToStaticMarkup(
+  /* eslint-disable no-console */
+  const originalConsoleError = console.error;
+  console.error = function (error, ...args) {
+    if (
+      error &&
+      !error.startsWith('Warning: useLayoutEffect does nothing on the server')
+    ) {
+      originalConsoleError(error, ...args);
+    }
+  };
+  const markup = renderToStaticMarkup(
```

Comment:
Since `renderToStaticMarkup` is used in a few other places beyond `getStoryMarkup`, it's best to use this to `packages/react`.

```js
import { renderToStaticMarkup as _renderToStaticMarkup } from 'react-dom/server';

/** @typedef {import('react').ReactElement} ReactElement */

/**
 * Render a React element to static markup.
 *
 * Silences any `useLayoutEffect` warnings.
 * 
 * Similar to `renderToString`, except this doesn't create extra DOM attributes
 * such as `data-reactid`, that React uses internally. This is useful if you want
 * to use React as a simple static page generator, as stripping away the extra
 * attributes can save lots of bytes.
 *
 * @param {ReactElement} element React element.
 * @return string Markup.
 */
function renderToStaticMarkup(element) {
	  const originalConsoleError = console.error;
	  console.error = function( error, ...args ) {
		    if ( error && ! error.startsWith( 'Warning: useLayoutEffect does nothing on the server' ) ) {
		      originalConsoleError( error, ...args );
		    }
	  };
	  
	  const markup = renderToStaticMarkup(...);
	  
	  console.error = originalConsoleError; // eslint-disable-line no-console

return markup;
}

export default renderToStaticMarkup;
```

And then `getStoryMarkup.js` just does:

```js
import { renderToStaticMarkup } from '@web-stories-wp/react`;
```